src/xcode/ENA/ENA/Source/HTTPClient/Resources/DCCRulesResource.swift

```diff
@@ -0,0 +1,132 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import HealthCertificateToolkit
+
+struct DCCRulesResource: Resource {
+
+	// MARK: - Init
+
+	init(
+		isFake: Bool = false,
+		ruleType: HealthCertificateValidationRuleType
+	) {
+		self.locator = .DCCRules(ruleType: ruleType, isFake: isFake)
+		self.type = .caching()
+		self.sendResource = EmptySendResource()
+		self.receiveResource = CBORReceiveResource<ValidationRulesModel>()
+		self.ruleType = ruleType
+	}
+
+	// MARK: - Overrides
+
+	// MARK: - Protocol Resource
+
+	typealias Send = EmptySendResource
+	typealias Receive = CBORReceiveResource<ValidationRulesModel>
+	typealias CustomError = DCCDownloadRulesError
+
+	var locator: Locator
+	var type: ServiceType
+	var sendResource: EmptySendResource
+	var receiveResource: CBORReceiveResource<ValidationRulesModel>
+
+	func customError(for error: ServiceError<DCCDownloadRulesError>) -> DCCDownloadRulesError? {
+		switch error {
+		case .transportationError:
+			return .NO_NETWORK
+		case .unexpectedServerError(let statusCode):
+					switch statusCode {
+					case (400...499):
+						return .RULE_CLIENT_ERROR(ruleType)
+					default:
+						return .RULE_SERVER_ERROR(ruleType)
+					}
+		case let .resourceError(resourceError):
+			return handleResourceError(resourceError)
+		default:
+			return nil
+		}
+	}
+
+#if !RELEASE
+	var defaultMockLoadResource: LoadResource? = {
+		guard let validationRules = try? ValidationRulesModel(decodeCBOR: HealthCertificateToolkit.rulesCBORDataFake()) else {
+			fatalError("broken cbor test data")
+		}
+		return LoadResource(result: .success(validationRules), willLoadResource: nil)
+	}()
+#endif
+
+	// MARK: - Private
+
+	private let ruleType: HealthCertificateValidationRuleType
+
+	private func handleResourceError(_ error: ResourceError?) -> DCCDownloadRulesError? {
+		guard let error = error else {
+			return nil
+		}
+		switch error {
+		case .missingCache:
+			return .RULE_MISSING_CACHE(ruleType)
+		case .missingData, .packageCreation:
+			return .RULE_JSON_ARCHIVE_FILE_MISSING(ruleType)
+		case .decoding(let dError):
+			if let ruleValidationError = dError as? RuleValidationError {
```

Comment:
Can this ever happen? In which situation does the RestService pass a `RuleValidationError` inside this func?

Reply:
This can happen when the CBOR decoding inside the `ValidationRulesModel` fails:
https://github.com/corona-warn-app/cwa-app-ios/blob/d3feafff70f31847196b61763b3fa8a22ea0d40e/src/xcode/ENA/ENA/Source/HTTPClient/Resources/ReceiveResources/CBORReceiveResource.swift#L51

Reply:
Do we need to distinguish here because the type gets lost because of the throw?